kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/ErrorCollector.kt

```diff
@@ -2,49 +2,24 @@
 
 package io.kotest.assertions
 
-import java.util.Stack
-
-actual val errorCollector: ErrorCollector = ThreadLocalErrorCollector
-
-object ThreadLocalErrorCollector : ErrorCollector {
-
-   private val clueContext = object : ThreadLocal<Stack<Clue>>() {
-      override fun initialValue(): Stack<Clue> = Stack()
-   }
-
-   private val failures = object : ThreadLocal<MutableList<Throwable>>() {
-      override fun initialValue(): MutableList<Throwable> = mutableListOf()
-   }
-
-   private val collectionMode = object : ThreadLocal<ErrorCollectionMode>() {
-      override fun initialValue() = ErrorCollectionMode.Hard
-   }
-
-   override fun setCollectionMode(mode: ErrorCollectionMode) = collectionMode.set(mode)
-
-   override fun getCollectionMode(): ErrorCollectionMode = collectionMode.get()
-
-   override fun pushClue(clue: Clue) {
-      clueContext.get().push(clue)
+import kotlinx.coroutines.asContextElement
+
+actual val errorCollector: ErrorCollector get() = ThreadLocalErrorCollector.instance.get()
```

Comment:
Do you think it's a bit unusual to have the val with get() ?
What about just making this a function?

Reply:
It's an `actual` for an `expect val`. Conceptually, it behaves as a val. Would you prefer changing the entire `expect` declaration and its target implementations?

Reply:
Yeah, I think it's clearer if we're using this one as a function to have the expect be a function and on JS / Native return a singleton if needed.

Reply:
IntelliJ detects 37 usages within Kotest. Probably better to change this separately from this PR. User-created matchers might also be affected.

Reply:
Ok fair enough.